storage/ndb/clusterj/clusterj-test/src/main/java/testsuite/clusterj/UnloadSchemaTest.java

```diff
@@ -0,0 +1,230 @@
+  }
+
+  Session getSession(String db) {
+    if (db == null) {
+      return sessionFactory.getSession();
+    } else {
+      return sessionFactory.getSession(db);
+    }
+  }
+
+  void returnSession(Session s) {
+    if (useCache) {
+      s.closeCache();
+    } else {
+      s.close();
+    }
+  }
+
+  void closeDTO(Session s, DynamicObject dto, Class dtoClass) {
+    if (useCache) {
+      s.releaseCache(dto, dtoClass);
+    } else {
+      s.release(dto);
+    }
+  }
+
+  public static class FGTest extends DynamicObject {
+    @Override
+    public String table() {
+      return TABLE;
+    }
+  }
+
+  public void runSQLCMD(AbstractClusterJModelTest test, String cmd) {
+    PreparedStatement preparedStatement = null;
+
+    try {
+      preparedStatement = connection.prepareStatement(cmd);
+      preparedStatement.executeUpdate();
+      System.out.println(cmd);
+    } catch (SQLException e) {
+      test.error("Failed to drop table. Error: " + e.getMessage());
+      throw new RuntimeException("Failed to command: ", e);
+    }
+  }
+
+
+  class DataInsertWorker extends Thread {
+    private boolean run = true;
+    private boolean running = false;
+    private int startIndex = 0;
+    private int insertsCounter = 0;
+
+    DataInsertWorker(int startIndex) {
+      this.startIndex = startIndex;
+    }
+
+    @Override
+    public void run() {
+
+      while (run) {
+        Session session = getSession(defaultDB);
+        DynamicObject e = null;
+        boolean rowInserted = false;
+        try {
+          e = (DynamicObject) session.newInstance(FGTest.class);
+          setFields(e, startIndex++);
+          session.savePersistent(e);
+          closeDTO(session, e, FGTest.class);
+          insertsCounter++;
+          rowInserted = true;
+        } catch (Exception ex) {
+//          ex.printStackTrace();
+          System.out.println(ex.getMessage());
+        } finally {
+          if (!rowInserted) {
+            session.unloadSchema(FGTest.class);
+            session.close();
+            try {
+              Thread.sleep(5);
+            } catch (InterruptedException ex) {
+              throw new RuntimeException(ex);
+            }
+          } else {
+            returnSession(session);
+          }
+        }
+      }
+      running = false;
+    }
+
+    public void stopDataInsertion() {
+      run = false;
+    }
+
+    public int getInsertsCounter() {
+      return insertsCounter;
+    }
+
+    public void setFields(DynamicObject e, int num) {
+      for (int i = 0; i < e.columnMetadata().length; i++) {
+        String fieldName = e.columnMetadata()[i].name();
+        if (fieldName.equals("id")) {
+          e.set(i, num);
+        } else if (fieldName.equals("col_1")) {
+          e.set(i, num);
+        } else if (fieldName.equals("col_2")) {
+          e.set(i, Long.toString(num));
+        } else if (fieldName.equals("col_3")) {
+          e.set(i, new Long(num));
```

Comment:
This doesn't compile on Mac OS X
I did instead:
long num_long = num
e.set(i, num_long);